### ISSUE TITLE
admin/console.html (js api) and flat parameter usage

### ISSUE THREAD

**Issue by raindog** [opensourceBIM/bimvie.ws#59]
Hi! The request  below: 
```
{
  "token": "myToken",
  "request": {
    "interface": "Bimsie1LowLevelInterface", 
    "method": "getDataObjectsByType", 
    "parameters": {
      "roid": "524291",
      "packageName": "ifc2x3tc1",
      "className": "IfcElementQuantity",
      "flat": "false"
    }
  }
}
```
done with the admin/console.html interface against a 1.4.0-FINAL BimServer raises this exception 
```
{
      "__type": "ServerException",
      "message": "Unknown error: null"
    }

```
If I set the "flat" parmeter to "true" then I get the correct response instead.
* Is this the intended behaviour? 
* Shall I set the flat parameter to false only to get the IfcQuantity* values? 
* How am I supposed to use the  /opensourceBIM/Tools/Tools/psd/IfcProductExtension/Pset_QuantityTakeOff.xml file?

TIA

**Comment by rubendel** [comment]
- This is not the intended behaviour, it's probably a bug in that version of BIMserver
- When flat=true if will only load the "oid", "name", "guid" and "type" fields. When flat=false it shoudl also load the other available fields
- Just tested with flat=false, it works here, but it might be a model-specific bug. Can you share your model? Also maybe this bug is already fixed in 1.5, can you try 1.5?
- The PSET documents have been copied from the IFC specs and were used to generate an excel file with schema info. The tools project is mainly a place to archive these one-time projects, you can use the code as an example for something you want to do, but most of the projects are not usable tools

**Comment by raindog** [comment]
Apparently I have this issues with various ifc files. See  [Project 0.9739047077018768.2.ifc.zip](https://github.com/opensourceBIM/bimvie.ws/files/484780/Project.0.9739047077018768.2.ifc.zip) for an example. I can manage to test this thing with the  1.5 version in the afternoon.
TIA

**Comment by raindog** [comment]
Ruben, the issues seems fixed on the version 1.5.56 of the server. With the flat flag on, it takes some time to get all the IfcElementQuantity items, but the operation completes.

I must stay on the 1.4 branch for now, but with a little more work, I should get the same result.
Thanks.

**Comment by rubendel** [comment]
Just tested the file you sent, it's working on 1.5. It's actually the formatting of the resulting JSON (about 1.5M) that takes a long time. Just pushed a fix for this in the Console project (it will just skip the formatting for larger results).
